File: minseok/2156/2156.cpp
#include <iostream>
#include <algorithm>

#define Max 100000

using namespace std;

int main(){
    int T;
    int wine[Max] = {0,};
    int dp[Max] = {0,};
    cin >> T;
    for(int i = 1; i<=T ; i++){
        cin >> wine[i];
    }

    dp[0] = wine[0];
    dp[1] = wine[1];
    dp[2] = wine[1] + wine[2];
    for(int i = 3 ; i<=T ; i++){
        dp[i] = max(dp[i-3]+wine[i-1]+wine[i],max(dp[i-2]+wine[i], dp[i-1]));
    }
    cout << dp[T];
}

File: yeonghwan/algorithm-boj/week3/2156.py
N = int(input())
wine = [int(input()) for _ in range(N)]
dp = [0] * N

if N > 2:
    dp[0], dp[1] = wine[0], wine[0] + wine[1]
    dp[2] = max((dp[0] + wine[2]), dp[1], (wine[1] + wine[2]))

    for i in range(3, N):
        dp[i] = max(dp[i-1], dp[i-2] + wine[i], dp[i-3] + wine[i-1] + wine[i])

    print(dp[-1])
elif N == 1:
    print(wine[0])
else:
    print(wine[0] + wine[1])
